### PR TITLE
refactor and signup validation

### PR DESCRIPTION
**Pull request by PuneetP16**
Mostly refactor.
Added  password validation in Sign Up page.

Files to Check
note
noteListing
FilterItem
FilterPanel

### CODE REVIEW

File: src/components/FilterItem/Filteritem.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+
+export const Filteritem = ({ dispatch, value, options }) => {
+	return (
+		<section className="fitler_type__wrapper">
+			<h3 className="h6 filter__head">Date</h3>
+
```
Comment (manojsarna) [inline src/components/FilterItem/Filteritem.jsx]: Suggestion : in the component there is a common heading Date

File: src/utilities/camelCaseWordGenerator.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export const camelCaseWordGenerator= (word) => {
+	return `${word.split("")[0].toUpperCase()}${word
+		.split("")
+		.splice(1)
+		.join("")}`;
```
Comment (manojsarna) [inline src/utilities/camelCaseWordGenerator.js:5]: good work on creating camelCase function

File: src/components/Note/Note.jsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	note,
+	togglePinBtn,
+	setColor,
+	archiveToggleFromNoteList,
+	toggleArchiveBtn,
+	deleteToggleFromList,
+	toggleTrashBtn,
+	editNote,
+}) => {
+	console.log(note);
```
Comment (gauravsinhaweb) [inline src/components/Note/Note.jsx]: remove all the console logs!

File: src/pages/Authentication/SignUp/SignUp.jsx [hunk 4]
```diff
@@ -40,20 +47,39 @@ export const SignUp = () => {
...
+			signUp({
+				signUpData,
+				navigate,
+				loginData,
+				userData,
+				dispatch,
+			});
+		} else {
+			console.log("error");
+			setAlert((a) => ({
```
Comment (manojsarna) [inline src/pages/Authentication/SignUp/SignUp.jsx:70]: Suggestion : Remove all console.log before submissions

File: src/pages/Authentication/SignUp/SignUp.jsx [hunk 5]
```diff
@@ -40,20 +47,39 @@ export const SignUp = () => {
...
+				...a,
```
Comment (gauravsinhaweb) [inline src/pages/Authentication/SignUp/SignUp.jsx:71]: use proper naming for readability (a).

**Review by gauravsinhaweb — APPROVED** [review]

**Review by manojsarna — APPROVED** [review]
overall its good


Do check the comments

File: src/pages/Authentication/SignUp/SignUp.jsx (hunk 4 above)
Comment (PuneetP16) [reply src/pages/Authentication/SignUp/SignUp.jsx:70]: Thanks for review will do the mentioned changes

File: src/components/FilterItem/Filteritem.jsx (hunk 1 above)
Comment (PuneetP16) [reply src/components/FilterItem/Filteritem.jsx]: Fixed it

File: src/pages/Authentication/SignUp/SignUp.jsx (hunk 5 above)
Comment (PuneetP16) [reply src/pages/Authentication/SignUp/SignUp.jsx:71]: will update it

File: src/components/Note/Note.jsx (hunk 3 above)
Comment (PuneetP16) [reply src/components/Note/Note.jsx]: noted!

**Pull request merged**